src/client/puppet-service.ts

```diff
@@ -582,6 +582,7 @@ export class PuppetService extends Puppet {
      *  cancel() is needed when grpc connection breaks
      */
     if (!this.grpcClient) {
+      log.warn('PuppetService', 'stopGrpcStream() trying to stopGrpcStream with no grpcClient')
```

Comment:
ditto

Reply:
I do think this should be a warning because close the eventStream when grpcClient is undefined means something is wrong elsewhere. This code tries to fix it but a warning should be raised.

Reply:
I understand that you are trying to debug this problem and warning seems reasonable to you. But image you are using some package developed by someone else, like `nestjs`, and you see some logs explaining some variables inside `nestjs` is wrong, what can you do? Would you like to see that?

Any logs that is higher than `verbose` should be something that we want the developers to know. Something like `puppet service started`, `puppet service is reconnecting` etc, for those logs, developer know what happened and what to do, but here, I don't think developers would know how to resolve this.